src/components/formio/textarea.tsx

```diff
@@ -37,6 +39,13 @@ export const TextArea: React.FC<JSX.IntrinsicElements['textarea'] & TextAreaProp
   const inputRef = useRef<HTMLInputElement>(null);
   const {bareInput} = useContext(RenderContext);
 
+  useLayoutEffect(() => {
+    if (autoExpand && inputRef.current) {
+      inputRef.current.style.height = 'inherit';
+      inputRef.current.style.height = `${inputRef.current.scrollHeight}px`;
+    }
+  }, [value]);
```

Comment:
```suggestion
  }, [autoExpand, value]);
```
I still need to look into setting up eslint in this project, but if `autoExpand` toggles between true/falls then this hook also needs to re-evaluate :)

Additionally, I think that `inputRef` should also be passed to the dependencies (but it has a stable identity).

Reply:
> Additionally, I think that `inputRef` should also be passed to the dependencies (but it has a stable identity).

Isn't using the value sufficient? What changes on `inputRef` could require the hook to be triggered?

Reply:
it's definitely something that eslint will complain about - and technically it *can* change even though it's a ref, consider this contrived example:

```tsx
const [value, setValue] = useState('foo');
const ref1 = useRef(null);
const ref2 = useRef(null);

let hookRef = ref1;
if (value === 'bar') {
  hookRef = ref2;
}

useLayoutEffect(
  () => {...},
  [hookRef],
);
```

So you can have the 'target' ref change for a particular hook depending on some other state or props.

This is why eslint essentially enforces that anything that is inside your hook body should be in the dependency array. It's also robustness for future changes in your code - ignoring the linter rule is dangerous since future requirements may require changes to the body of the hook and then it's easy to forget to add any relevant new dependencies.

In this case, the ref indeed does not change, so it will *also* not trigger a new evaluation of the hook body, meaning the end result is the same and you'll be able to satisfy the linter.